lib/truffle/thread.rb

```diff
@@ -29,4 +29,21 @@ class ThreadError < StandardError
 class Thread
   Queue = ::Queue
   SizedQueue = ::SizedQueue
+
+  class Queue
+    def pop(non_block = false, timeout: nil)
+      if non_block
+        raise(ArgumentError, "can't set a timeout if non_block is enabled") unless Primitive.nil?(timeout)
```

Comment:
A minor style thing, but raise and alias_method below are considered pseudo keywords in Ruby, so are usually not used with parentheses